temporalcli/commands.go

```diff
@@ -561,3 +562,17 @@ func timestampToTime(t *timestamppb.Timestamp) time.Time {
 type nopWriter struct{}
 
 func (nopWriter) Write(b []byte) (int, error) { return len(b), nil }
+
+type structuredError struct {
+	Message string `json:"message"`
+	Type    string `json:"type,omitempty"`
+	Details any    `json:"details,omitempty"`
+}
+
+func fromApplicationError(err *temporal.ApplicationError) *structuredError {
+	return &structuredError{
+		Message: err.Error(),
+		Type:    err.Type(),
+		Details: err.Details(),
```

Comment:
`.Details` expects a pointer to deserialize into and returns an error if it can't. You could have `var details any` and then `err.Details(&details)` (checking error) and then set `details` to in this field

Reply:
Ah thanks good catch